### ISSUE TITLE
Publish as Community Plugin

### ISSUE THREAD

**Issue by lukemt** [aidan-gibson/obsidian-opener#17]
I think it's about time to publish the plug-in for the community. It may not be perfect, though it's already pretty useful. 

Any critical bugs or features missing for the release? What do you think?

As far as automated testing goes, maybe we can start by writing down test cases and use them for manual tests. Fully automated testing seems a bit excessive to me (although cool 😎)

**Comment by aidan-gibson** [comment]
Fuck it we ball 😤😤😤

**Comment by aidan-gibson** [comment]
I remember there being some config option somewhere to specify DesktopOnly...honestly I think just that and we're good. 

Not sure how fully automated testing would work anyway, it's just a bit of a drag to test all the different settings options with every minor change pushed. But I should test everything manually before publishing for sure(:

**Comment by lukemt** [comment]
Hell yeah!

**Comment by lukemt** [comment]
Hell yeah! Preparing the Release...

### Thoughts on Tab Navigation Behavior

I was thinking, should there be a way to navigate to a different file and replace the current tab, just like the default behavior, for example, by clicking a link? But wait, alt-clicking already has another function, so that might not be a good idea. Someone else suggested an interesting twist – command click to replace a tab, and normal click to open a new one. Honestly, not a fan of that as it inverts the standard functionality, which is kind of weird. For me, I don't miss that feature at all. If I want to replace a tab, I can just open a new one and close the old.  Probably, most people installing this wouldn't bother too much. So, my conclusion is, it's fine the way it is, though we might consider this as an optional additional setting or just keep thinking about better ideas.

### On Mobile Compatibility

Now, about this mobile thing. Currently, the plugin is desktop-only, and that's cool for now. isDesktopOnly is set in manifest.json already. But you know what? Let's think ahead. We can start by activating this thing on mobile and see how it goes. Maybe, we can implement a feature flag for optional activation on mobile. You could have a setting to switch on. Set to false initially, so manual activation is needed. Yeah, that could be a new feature. We can create a new issue for that, but for now, it's alright to be desktop-only. 

### On Handling PDF Files

Okay, let's talk about PDFs and other files. I don't use 'em much in Obsidian, but when I do, I'd probably prefer opening them in Obsidian. Although I'd like to have the option of both.  So, my preference would be this: When I click on a PDF link, it opens within Obsidian, but in a new tab. But when I command click on the PDF link, it opens in the native app. Best of both worlds!

**Comment by lukemt** [comment]
Hi @aidan-gibson 

when you find the time maybe create a new version :P

**Comment by aidan-gibson** [comment]
Hi Luke! I'm down for as many options as you'd like. That sounds fine to me. I strongly preferred opening pdfs in Preview prior, but Obsidian has really upped their game with the native pdf reader, so I'm now cool with defaulting to Cmd+Click to open via system default pdf reader. 

I'm a bit hesitant to publish to community plugins while #23 is still looming large, but pending that bug getting squashed I'm ready to roll.

**Comment by aidan-gibson** [comment]
@lukemt I sent you an invite as a collaborator btw! Sorry I didn't do so earlier, I've never done this before and wasn't sure how it worked.
To trigger the github action for new releases, after making commits, you do the following
`npm version patch -m "<version message here>" && git push origin --tags && git push` (I know the extra git push is totally cringe, haven't had time to uncringify it yet lmao). Thanks again for all the time you've put into this!

**Comment by lukemt** [comment]
perfect!

**Comment by austin-joseph** [comment]
Whats required to publish this as a community plugin? I see that #23 is closed. Id be happy to step in and do the work. Ive tried other plugins but none offer the functionality I need like this one does.

**Comment by lukemt** [comment]
Hi @austin-joseph , Thank you for your appreciation and your offer to help. I just tested the current implementation and it seems quite broken. Opening a tab thats already open, does not work at all anymore. Previously it switched to that tab. Don't you have that issue?

**Comment by lukemt** [comment]
ok i got it to work using the zwergziege patch. still need to figure out how to fix the templater issues

**Comment by austin-joseph** [comment]
I do have the issue but its closer to minor annoyance rather than show stopping problem. 
I also have the issue with templater mentioned in #57 if thats what youre referencing

At the end of the day id rather have those issues than the default obsidian behavior.

What is the zwergziege patch youre talking about? Im currently getting familiar with the source so I can try to get to a root cause on #57

**Comment by lukemt** [comment]
@austin-joseph thank you. I meant this guy #63 

It does work for me now, do you still face any issues?

Yeah that's exactly the templater problem i was referring to.

**Comment by austin-joseph** [comment]
No I don't face any issues with new tabs running the plugin from main.

In the meantime while a fix for #57 is developed can we get a minor release to push the #63 to users?

**Comment by lukemt** [comment]
Hi @austin-joseph ,

That's good news.

I've already fixed #57 which means it is officially bug-free.

I will continue development over at my fork and suggest we continue from there.

I will bundle the new version asap.

**Comment by austin-joseph** [comment]
Is your fork public? Scratch that i found it.

If you still have rights, update the README in this project to point to your fork in order to be more accurate. Mononote is not a drop in replacement for this plugin.

Thank you for taking the time to work on this

**Comment by lukemt** [comment]
Yeah, unfortunately I do not seem to have any rights here anymore.

Maybe @aidan-gibson can help out?

Here is the link to the repo:
https://github.com/lukemt/obsidian-opener

I'm still thinking of a speaking new name. Anyways here we go, I finally fixed the build. Is public now.
